### PR TITLE
Add option to use `time` crate instead of `chrono`

### PR DESCRIPTION
**Pull request by utterstep**
This is useful for users who are already working with `sqlx` and `time`, bc `sqlx` doesn't allow you to use `chrono` and `time` together (I have such a case, for example :)).

This PR proposes non-breaking change to allow such users specify that they need `time`, not `chrono`

* leaves the default as `chrono` but allows users to opt-in to using `time` instead
* adds a `time` feature to `apalis-core`, `apalis-sql` and `apalis-redis`
* enabling both `cron` and `time` features now triggers a compile error because `apalis-cron` uses `cron` and `cron` depends on `chrono`
* adds `time` to the CI matrix
* (subjective) use `datetime - duration` instead of `datetime.sub(duration)`

I'd gladly accept any input from your side on how to make my proposal work for you to include it to the upstream :)

### CODE REVIEW

**Comment by utterstep** [comment]
Hi! Thanks for your attention to this PR!

I've updated tests, can you restart CI at some point? :)

**Comment by utterstep** [comment]
Fixed rustfmt errors (🤦)

**Comment by utterstep** [comment]
@geofmureithi all checks are passing now, what would you say about merging this PR?

Just to clarify – I'm absolutely OK to make any adjustments you'd like, regarding API/docs/examples/... :)

**Comment by geofmureithi** [comment]
> what would you say about merging this PR?

The main reason this is pending is that I have been trying to move away from feature flags towards using traits.
Traits allow someone to decide what crate to use instead of hardcoding.
While I appreciate the PR, I wish we had discussed it before you did all this work.

Would you be ok if this PR is worked on into v0.5? We can start making those changes and start a 0.5-alpha branch.

File: packages/apalis-cron/src/lib.rs [hunk 1]
```diff
@@ -69,6 +69,9 @@
 //! }
 //! ```
 
+#[cfg(feature = "time")]
```
Comment (geofmureithi) [inline packages/apalis-cron/src/lib.rs:72]: Is there a point of adding a time feature in apalis-cron?

**Comment by utterstep** [comment]
> Would you be ok if this PR is worked on into v0.5? We can start making those changes and start a 0.5-alpha branch.

Sure, no problem!

> The main reason this is pending is that I have been trying to move away from feature flags towards using traits.
Traits allow someone to decide what crate to use instead of hardcoding.

Makes a lot of sense to me! However, there is one catch, as I see it now: `sqlx` supports `time`/`chrono` via mutually exclusive features currently (to be more precise, if both features are 'on' the `chrono` takes precedence silently, breaking all `time`
-expecting usage).
So `apalis` need to allow selection of `sqlx` features at least, and I don't know if that's possible using the traits only, seems like it's not currently :(

File: packages/apalis-cron/src/lib.rs (hunk 1 above)
Comment (utterstep) [reply packages/apalis-cron/src/lib.rs:72]: I'm not sure :)
My main reasoning was to inform anyone enabling both `cron` and `time` features of their incompatibility as earlier as possible.

**Comment by geofmureithi** [comment]
> So apalis need to allow selection of sqlx features at least, and I don't know if that's possible using the traits only, seems like it's not currently :(

I am actually thinking of dropping `sqlx` for `rbatis`

**Review by geofmureithi — APPROVED** [review]

**Comment by utterstep** [comment]
@geofmureithi thank you so much for your collaboration! :)

**Pull request merged**